GiEnJul/ClientApp/src/components/alert/AlertHover.tsx

```diff
@@ -0,0 +1,71 @@
+import { Alert, AlertProps } from '@material-ui/lab';
+import React, { FC } from "react";
+import { useState } from 'react';
+import { useEffect } from 'react';
+import useStyles from './Styles';
+
+
+interface IAlertHover extends AlertProps {
+    fadeTime?: number,  // Defaults to 4000 ms
+    className?: string,
+    children?: React.ReactNode,
+
+    // Use setChildren property to reset the childen after hiding the alert. 
+    // This function run 400 ms after the alert starts to hide.
+    // This property is necessary for rerendering the alert with the same message.
+    // Throws an error if the children property is set, but not the setChildren property.
+    setChildren?: React.Dispatch<React.SetStateAction<React.ReactNode>>,
+    
+    id?: string,
+    onClose?: (event: any) => void,
+}
+
+const AlertHover: FC<IAlertHover> = ({
+    fadeTime,
+    className,
+    children,
+    setChildren,
+    id,
+    onClose,
+    ...rest
+}) => {
+    const classes = useStyles();
```

Comment:
useStyels entales this is a hook, however i only see classes specifically for this component. ( i may be wrong) However it doesnt seem necessary to write it separately unless youll be exporting different classses for different components from it. We can go over this.